ODS (OpenDocument Spreadsheet) Exporter for EMF

ideas-into-software opened the pull request:
Signed-off-by: Michael H. Siemaszko <m.siemaszko@datainmotion.com>

Commit:
ODS (OpenDocument Spreadsheet) Exporter for EMF

Signed-off-by: Michael H. Siemaszko <m.siemaszko@datainmotion.com>

juergen-albert commented:
Looks good in general. @stbischof and I found some minor things though:

* Autosize coloums if possible
* References:
    * Many References via Mapping table in an extra Sheet
    * Ref Header as e.g. ref_father
    * if Ref Type has an id use the `EMFUtil.getID` value
    * else create a pseudo ID and use this
* Metadata Sheets: Documentation is missing
* Remove the E from Ecore Datatypes e.g. EString -> String
* Metadata for Enum Types are missing

Commit:
ODS (OpenDocument Spreadsheet) Exporter for EMF (c.d.)

 - Specialized EMF ODS resource

 - References traceable via IDs and (optionally) links

 - Additional export options

 - Additional metadata

 - Documentation

 - Various fixes & improvements


Signed-off-by: Michael H. Siemaszko <m.siemaszko@datainmotion.com>

juergen-albert commented:
Can you add the new project here https://github.com/geckoprojects-org/org.gecko.emf.utils/blob/438236f69d233f46444bd066c1d68b43e6952763/org.gecko.emf.util.bnd.library.workspace/required.bndrun and resolve it? With this it will become part of the library.

ideas-into-software commented:
Please see https://github.com/geckoprojects-org/org.gecko.emf.utils/pull/8 for replacement - i.e. due to licensing issue with FastODS, I have replaced it with SODS (https://github.com/miachm/SODS)
